enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -0,0 +1,378 @@
+ `ConfigMaps` during their Pod's execution.
+ 
+The precise means of opting into consuming the shared content are outside the scope of this proposal.
+
+This proposal was broken off from [a proposal on injecting RHEL entitlements/subscriptions](https://github.com/openshift/enhancements/pull/214)
+**EDITORIAL NOTE:** if/when PR 214 merges the reference link will be adjusted to point to the merged document hosted on
+https://github.com/openshift/enhancements.  Or if that PR is replaced by another, this proposal will point to that.
+
+
+## Motivation
+
+Help simplify consumption of `Secrets` and `ConfigMaps` defined in one namespace by pods in other namespaces by providing
+the foundational pieces for building a shared secret/configmap feature such that user pods in one namespace can consume
+secrets/configmaps from another namespace (which is presumably maintained by a cluster admin).
+
+### Goals
+
+Provide easy to use sharing of `Secrets` and `ConfigMaps` defined in one namespace by pods in other namespaces.
+
+### Non-Goals
+
+1. This proposal assumes the sensitive content encapsulated in the `Secret` or `ConfigMap` is valid and correct for the user's purposes.
+No validation of the content for specific usage within a Pod will be done.  That said, this proposal does provide 
+for watching for updates to those objects.
+So if the cluster administrator realized on their own they have provided bad data, and then corrects it, this proposal
+will watch for, discover, and propagate the update when new Pods consuming the content are provisioned.
+
+2. On protecting this content from being viewed/copied by users: if it's available for the Pod to use, the user who can 
+create a Pod that can utilize this feature can also read / view / exfiltrate the Secret contents. It is technically 
+impossible to avoid that.
+
+3. Providing security constraints on who can create these volumes and/or use them in their pods to access secrets/configmaps
+from another namespace.  That will be addressed in a separate proposal as noted in the Summary. 
+
+
+## Proposal
+
+### User Stories [optional]
+
+1. As a cluster admin with an OpenShift subscription, I want to share the entitlement keys and configuration with my 
+users' workload Pods so that containers in those Pods can install RHEL subscription content.
+
+2. As a cluster admin with a set of CAs pertinent to various users's workloads, I want to share those CAs with my 
+users' workload Pods so that containers in those Pods can utilize those CAs when performing TLS based communication.
+
+3. See [this proposal](https://github.com/openshift/enhancements/pull/214) for additional user scenarios that could 
+utilize this proposal.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Install
+
+A comparison of installing the implementation of this proposal via OLM vs. CVO was done in another enhancement proposal.  
+The net result was to go with CVO.
```

Comment:
This result surprises me.  Such a CSI plugin has immense value for the community and is a fairly open space for innovation.  I would hope to see this distributed via OLM first and by the CVO second (if at all).

Reply:
definitely a point for on-going discussion, but absent a way to have OLM operators installed on day one, making this an OLM operator really hamstrings our ability to deliver the feature we are ultimately building this for, seamless consumption of entitlements.

Reply:
I'll re-add this as an open question

It had been an open question until I saw the exchange @deads2k between you and @jsafrane in https://github.com/openshift/enhancements/pull/139/files/1ff25b741829deac9e6ba3826797c31a2f3430df#r391616151 and https://github.com/openshift/enhancements/pull/139/files/1ff25b741829deac9e6ba3826797c31a2f3430df#r391761362

I'm taking your comment here that my interpretation that a consensus was reached was incorrect.

Reply:
@gabemontero i think we probably need @derekwaynecarr to help us make a decision on this one.

Reply:
How far off are we from being able to have another operator (or even the CVO) require an OLM operator to be present?  I think that needs to be an objective in order to migrate any operator to an OLM based one.